core/database/src/main/java/com/sergius/core/database/RoomLocalAgendaDataSource.kt

```diff
@@ -76,25 +83,40 @@ class RoomLocalAgendaDataSource(
         }
     }
 
-    override fun getReminders(): Flow<List<Reminder>> {
-        return reminderDao.getReminders().map { it.map { it.toReminder() } }
+    override fun getRemindersForDate(date: LocalDate): Flow<List<Reminder>> {
+        val start = startOfDay(date)
+        val end = endOfDay(date)
+        return reminderDao.getReminders(
+            dateFrom = start,
+            dateTo = end
+        ).map { it.map { it.toReminder() } }
     }
 
+    private fun startOfDay(date: LocalDate) = date
+        .atStartOfDay(ZoneId.systemDefault())
+        .toInstant()
+        .toEpochMilli()
+
+    private fun endOfDay(date: LocalDate) = date
+        .plusDays(1)
+        .atStartOfDay(ZoneId.systemDefault())
+        .minusSeconds(1)
```

Comment:
Suggestion: you could also use `LocalTime.MAX` here. It’s not necessarily cleaner, but it makes the intent (“end of day”) explicit without doing `plusDays(1).minusSeconds(1)` 😊:

```kotlin
private fun endOfDay(date: LocalDate) = date
    .atTime(LocalTime.MAX) // 23:59:59.999999999
    .atZone(ZoneId.systemDefault())
    .toInstant()
    .toEpochMilli()
```